common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -219,25 +219,56 @@ def selected_children(self):
         if hasattr(self, "_selected_set"):
             # Already done:
             return self._selected_set  # pylint: disable=access-member-before-definition
+
+        lib_tools = self.runtime.service(self, 'library_tools')
+        format_block_keys = lambda block_keys: lib_tools.create_block_analytics_summary(self.location.course_key, block_keys)
+
         # Determine which of our children we will show:
         selected = set(tuple(k) for k in self.selected)  # set of (block_type, block_id) tuples
```

Comment:
Is `self.selected` the current set in the library or the current set in the course using the library? That is, is this code removing existing things from the course b/c they disappeared from the library (which I'm assuming)?